extensions/common/iam/identity-trust/identity-trust-core/src/main/java/org/eclipse/edc/iam/identitytrust/core/IatpParticipantAgentServiceExtension.java

```diff
@@ -0,0 +1,47 @@
+/*
+ *  Copyright (c) 2024 Amadeus
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Amadeus - initial API and implementation
+ *
+ */
+
+package org.eclipse.edc.iam.identitytrust.core;
+
+import org.eclipse.edc.identitytrust.model.CredentialSubject;
+import org.eclipse.edc.identitytrust.model.VerifiableCredential;
+import org.eclipse.edc.spi.agent.ParticipantAgentServiceExtension;
+import org.eclipse.edc.spi.iam.ClaimToken;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Map;
+import java.util.Objects;
+
+import static java.util.Collections.emptyMap;
+import static org.eclipse.edc.iam.identitytrust.core.IatpDefaultServicesExtension.CLAIMTOKEN_VC_KEY;
+import static org.eclipse.edc.spi.agent.ParticipantAgent.PARTICIPANT_IDENTITY;
+
+/**
+ * Retrieve subject id from the list of {@link VerifiableCredential} and set the
+ * PARTICIPANT_IDENTITY attribute accordingly.
+ */
+public class IatpParticipantAgentServiceExtension implements ParticipantAgentServiceExtension {
+    @Override
+    public @NotNull Map<String, String> attributesFor(ClaimToken token) {
+        return token.getListClaim(CLAIMTOKEN_VC_KEY)
+                .stream()
+                .map(o -> (VerifiableCredential) o)
```

Comment:
minor nit: having a `filter(o -> o instanceOf VerifiableCredential)` makes this a bit more robust and guards against potential class-cast exceptions